gcd/src/main/java/org/spine3/server/storage/datastore/DatastoreProperties.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright 2015, TeamDev Ltd. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.spine3.server.storage.datastore;
+
+import com.google.api.services.datastore.DatastoreV1;
+import com.google.protobuf.FieldMask;
+import com.google.protobuf.Message;
+import com.google.protobuf.TimestampOrBuilder;
+import org.spine3.base.EventContextOrBuilder;
+import org.spine3.protobuf.Messages;
+import org.spine3.server.storage.EventStorageRecordOrBuilder;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Date;
+
+import static com.google.api.services.datastore.client.DatastoreHelper.makeProperty;
+import static com.google.api.services.datastore.client.DatastoreHelper.makePropertyReference;
+import static com.google.api.services.datastore.client.DatastoreHelper.makeValue;
+import static org.spine3.protobuf.Timestamps.convertToDate;
+import static org.spine3.protobuf.Timestamps.convertToNanos;
+
+/**
+ * Utility class, which simplifies creation of datastore properties.
+ *
+ * @author Mikhail Mikhaylov
+ */
+@SuppressWarnings("UtilityClass")
+/* package */ class DatastoreProperties {
+
+    private static final String TIMESTAMP_PROPERTY_NAME = "timestamp";
+    /* package */ static final String TIMESTAMP_NANOS_PROPERTY_NAME = "timestamp_nanos";
+
+    /* package */ static final String AGGREGATE_ID_PROPERTY_NAME = "aggregate_id";
+    private static final String PRODUCER_ID_PROPERTY_NAME = "producer_id";
+    /* package */ static final String EVENT_TYPE_PROPERTY_NAME = "event_type";
+    private static final String EVENT_ID_PROPERTY_NAME = "event_id";
+
+    private static final String CONTEXT_FIELD_PROPERTY_PREFIX_NAME = "context_";
+
+    private static final String CONTEXT_EVENT_ID_PROPERTY_NAME = "context_event_id";
+    private static final String CONTEXT_TIMESTAMP_PROPERTY_NAME = "context_timestamp";
+    private static final String CONTEXT_COMMAND_CONTEXT_PROPERTY_NAME = "context_command_context";
```

Comment:
`context_command_context` looks strange. We can try `context_of_command` if you need the prefix.